### PR TITLE
Deploy on pathfinder using a tools image (using asdf)

### PR DESCRIPTION
**Pull request by matthieu-foucault**
+ explicitely committed nodejs relase team keyring

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+USER 1001
+
+# Because asdf is loaded via BASH_ENV, all commands using adsf need to be executed using /usr/bin/env bash -c
+RUN /usr/bin/env bash -c "cat ${APP_ROOT}/.tool-versions | cut -f 1 -d ' ' | xargs -n 1 asdf plugin-add"
+RUN /usr/bin/env bash -c "asdf plugin-update --all"
+# The nodejs release team keyring is needed to install the asdf node plugin
+COPY ./.bin/import-nodejs-keyring.sh /tmp/import-nodejs-keyring.sh
+RUN /tmp/import-nodejs-keyring.sh
+
+RUN /usr/bin/env bash -c "POSTGRES_EXTRA_CONFIGURE_OPTIONS='--with-libxml' asdf install"
```
Comment (wenzowski) [inline Dockerfile]: `ENV POSTGRES_EXTRA_CONFIGURE_OPTIONS='--with-libxml'` please. That way a consumer of this image can still `asdf install` if a different version is needed

File: openshift/deploy/job/schema-revert.yml [hunk 2]
```diff
@@ -42,8 +42,7 @@ objects:
           - /usr/bin/env
           - bash
           - -c
-          - |
-            sqitch revert -y;
+          - sqitch revert -y || exit 0
```
Comment (wenzowski) [inline openshift/deploy/job/schema-revert.yml]: do we really want to ignore revert failures?

File: schema/.s2i/bin/assemble [hunk 3]
```diff
@@ -33,46 +19,7 @@ else
...
-if [ -f "cpanfile" ]; then
-  echo "---> Installing modules from cpanfile ..."
-  cpanm $MIRROR_ARGS $ENABLE_CPAN_TEST -l extlib Module::CoreList
-  # Installing Sqitch from url
-  cpanm $MIRROR_ARGS $ENABLE_CPAN_TEST -l extlib https://github.com/bcgov/cas-sqitch/releases/download/v1.0.1.TRIAL/App-Sqitch-v1.0.1-TRIAL.tar.gz
-  cpanm $MIRROR_ARGS $ENABLE_CPAN_TEST -l extlib --installdeps .
-else
-  echo "---> No cpanfile found, nothing to install"
-fi
+make install CPANM="cpanm $ENABLE_CPAN_TEST -l extlib"
```
Comment (wenzowski) [inline schema/.s2i/bin/assemble]: can haz comment plz

File: schema/.s2i/environment [hunk 4]
```diff
@@ -1,4 +1,6 @@
 PERL_MB_OPT=--install_base "/opt/app-root/src/extlib/lib/perl5"
 PERL_MM_OPT=INSTALL_BASE=/opt/app-root/src/extlib/lib/perl5
 PERL5LIB=/opt/app-root/src/extlib/lib/perl5
-PATH=/opt/app-root/src/extlib/bin:/opt/app-root/src/bin:/opt/app-root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+# cpanm is installed in /opt/app-root/src/extlib/lib/perl5/bin
+# sqitch is installed in /opt/app-root/src/extlib/bin
+PATH=/opt/app-root/src/extlib/bin:/opt/app-root/src/extlib/lib/perl5/bin:/opt/app-root/src/bin:/opt/app-root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```
Comment (wenzowski) [inline schema/.s2i/environment]: 👍

File: schema/Makefile [hunk 5]
```diff
@@ -150,23 +148,23 @@ install_pgtap: pgtap
...
-	@@${CPAN} App:cpanminus
+	@@echo | $(CPAN) # accept cpan defaults blindly
+	@@$(CPAN) App:cpanminus
 endif
 
 .PHONY: install_cpandeps
 install_cpandeps:
 	# install sqitch
-	${CPANM} -n https://github.com/matthieu-foucault/sqitch/releases/download/v1.0.1.TRIAL/App-Sqitch-v1.0.1-TRIAL.tar.gz
+	$(CPANM) -n https://github.com/matthieu-foucault/sqitch/releases/download/v1.0.1.TRIAL/App-Sqitch-v1.0.1-TRIAL.tar.gz
```
Comment (wenzowski) [inline schema/Makefile]: are we still waiting for a release on this? If so, maybe we'll migrate to a bcgov repo

Comment (matthieu-foucault) [reply schema/Makefile]: My PR there has some comments that need to be taken care of in order for it to be merged. We're not using the feature that was developed anyway, so we can revert to sqitch 0.9999.

File: openshift/deploy/job/schema-revert.yml (hunk 2 above)
Comment (matthieu-foucault) [reply openshift/deploy/job/schema-revert.yml]: Currently revert will run on the last deploy job's commit, whether it succeeded or not. So yes. This should be fixed in the Makefile

**Pull request merged**